x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -629,6 +629,31 @@ func TestCalcJoinPoolShares(t *testing.T) {
 	}
 }
 
+func TestCalcJoinPoolSharesNoSwap(t *testing.T) {
+	// We append shared calcSingleAssetJoinTestCases with multi-asset and edge test cases.
+	// See calcJoinSharesTestCase struct definition for explanation why the sharing is needed.
+	testCases := append([]calcJoinSharesTestCase{}, calcSingleAssetJoinTestCases...)
```

Comment:
Multi-asset tests coming: https://github.com/osmosis-labs/osmosis/pull/1732

Reply:
Tests moved to: #1767